app/screens/localization/LanguageSelection.tsx

```diff
@@ -229,42 +230,49 @@ const LanguageSelection = ({ route, navigation }: Props): any => {
     } else {
       console.log(language, "..newLanguage");
       if (Object.keys(route.params).length !== 0) {
-        console.log(route.params.country?.CountryID, "routeparams is");
-        const filteredLan = route.params?.country?.languages?.filter((lang: any) => lang.languageCode == language.languageCode);
+        console.log(route.params?.country?.CountryID, "routeparams is");
+        const filteredLan = route.params?.country?.languages?.filter((lang: any) => lang?.languageCode == language?.languageCode);
         console.log('filtered language is', filteredLan)
         dispatch(onLocalizationSelect({ "languages": filteredLan, "countryId": route.params.country?.CountryID }));
         dispatch(setInfoModalOpened({ key: 'dailyMessageNotification', value: '' }));
-        analytics().setUserProperties({ country: route.params.country.name, language: language.displayName })
+        analytics().setUserProperties({ country: route.params?.country?.name, language: language?.displayName })
       } else {
         console.log(country, "countryData");
         dispatch(onLocalizationSelect(country));
         dispatch(setInfoModalOpened({ key: 'dailyMessageNotification', value: '' }));
-        analytics().setUserProperties({ country: country.displayName, language: language.displayName })
+        analytics().setUserProperties({ country: country?.displayName, language: language?.displayName })
       }
 
       // if (userIsOnboarded == true) {
       //   dispatch(setSponsorStore({ country_national_partner: null, country_sponsor_logo: null }));
       // }
       console.log('Sponsors Data for countryList', sponsorsData)
       dispatch(setSponsorStore(sponsorsData));
-      navigation.navigate('LoadingScreen', {
+      const params = {
         apiJsonData: userIsOnboarded == true ? allApisObject(false, incrementalSyncDT) : apiJsonData,
         prevPage: userIsOnboarded == true ? 'CountryLangChange' : 'CountryLanguageSelection'
-      });
+      }
+      navigation.navigate("LoadingScreen", params );
     }
   }
   const goToConfirmationScreen = (): any => {
-    let newLanguage: any = null
+    let newLanguage: any = null;
     if (language?.locale != undefined) {
-      newLanguage = language
+      newLanguage = language;
+    } else if (Array.isArray(language) && language.length > 0) {

```

Comment:
language?.length

Reply:
in this we have 2 condition if we get as object and as array we handled both ways